src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs

```diff
@@ -0,0 +1,286 @@
+ *
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading.Tasks;
+using Tizen.NUI.BaseComponents;
+using Tizen.NUI.Binding;
+
+namespace Tizen.NUI.Components
+{
+    /// <summary>
+    /// The Navigator is a class which navigates pages with stack methods such
+    /// as Push and Pop.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class Navigator : Control
+    {
+        //This will be replaced with view transition class instance.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected Animation _curAnimation = null;
+
+        //This will be replaced with view transition class instance.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected Animation _newAnimation = null;
+
+        /// <summary>
+        /// Creates a new instance of a Navigator.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public Navigator() : base()
+        {
+        }
+
+        /// <summary>
+        /// List of pages of Navigator.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public List<Page> NavigationPages { get; } = new List<Page>();
+
+        /// <summary>
+        /// Pushes a page to Navigator.
+        /// If the page is already in Navigator, then it is not pushed.
+        /// </summary>
+        /// <param name="page">The page to push to Navigator.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the argument page is null.</exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public void Push(Page page)
+        {
+            if (page == null)
+            {
+                throw new ArgumentNullException(nameof(page), "page should not be null.");
+            }
+
+            //Duplicate page is not pushed.
+            if (NavigationPages.Contains(page)) return;
+
+            var curTop = Peek();
+
+            if (!curTop)
+            {
+                Insert(0, page);
+                return;
+            }
+
+            NavigationPages.Add(page);
+            Add(page);
+
+            //Invoke Page events
+            page.InvokeAppearing();
+            curTop.InvokeDisappearing();
+
+            //TODO: The following transition codes will be replaced with view transition.
+            if (_curAnimation)
+            {
+                _curAnimation.Stop();
+                _curAnimation.Clear();
+            }
+
+            if (_newAnimation)
+            {
+                _newAnimation.Stop();
+                _newAnimation.Clear();
+            }
+
+            _curAnimation = new Animation(1000);
+            _curAnimation.AnimateTo(curTop, "Scale", new Vector3(0.0f, 0.0f, 1.0f), 0, 1000);
+            _curAnimation.AnimateTo(curTop, "Opacity", 0.0f, 0, 1000);
+            _curAnimation.Play();
+            _curAnimation.Finished += (object sender, EventArgs e) =>
+            {
```

Comment:
If you intended to reset properties after animation finished,
how about try this?
```C#
// When the animation ends, the animated property values are forgotten.
animation.EndAction = EndActions.Discard;
```

Reply:
Thank you for the information. I will check it and modify the code.
Thank you.